packages/nameguard-react/src/components/LoadingReportIcon/LoadingReportIcon.tsx

```diff
@@ -0,0 +1,57 @@
+import React from "react";
+
+import { Tooltip } from "../Tooltip/Tooltip";
+import { ENSName } from "@namehash/ens-utils";
+import { RatingIcon, RatingIconSize } from "../Report/RatingIcon";
+import { RatingLoadingIcon } from "../icons/RatingLoadingIcon";
+
+type LoadingReportIconProps = {
+  onClickOverride?: (ensName: ENSName) => void;
+  size?: RatingIconSize;
+} & React.ComponentProps<typeof RatingIcon>;
+
+export const LoadingReportIcon = ({
+  onClickHandler,
+  size = RatingIconSize.small,
+
+  /*
+    Props are applied to the shield icon which is the onHover trigger element 
+    for the tooltip with Report information. For examples, please visit the
+    https://nameguard.io/docs/report and see the ReportIcon docs. Any 
+    additional props are passed to the shield icon that when hovered,
+    displays the tooltip with the report information.
+  */
+  ...props
+}: LoadingReportIconProps) => {
+  return (
+    <Tooltip
+      trigger={RatingLoadingIcon({
+        size,
+        ...props,
+      })}
+    >
+      <div className="flex items-start space-x-3 py-2.5 max-w-[300px]">
+        <div className="mt-0.5">
+          <RatingLoadingIcon fill="#CFCFCF" size={RatingIconSize.small} />
+        </div>
+
+        <div className="flex-1">
+          <div className="flex items-center justify-between">
+            <span className="font-semibold mb-1 text-white">
+              Loading report...
+            </span>
+          </div>
+
+          <div className="text-sm text-white">
+            <button
+              className="appearance-none underline font-medium"
+              onClick={onClickHandler}
```

Comment:
Hmm.. I don't think this calls the callback correctly? How is the param getting passed to the callback?